### ISSUE TITLE
Provide Eq instance (getEq)

### ISSUE THREAD

**Issue by justin-hackin** [no-day/fp-ts-sized-vectors#1]
I'm very new to fp-ts so I hope I'm not overlooking something obvious here. I'm looking for how to implement a fp-ts compatible equality operation for the sized vectors. I tried this 

```ts
export type Point2D = Vec<2, number>;

export const point = (x: number, y: number): Point2D => vec(x, y);
const eqNumber: eq.Eq<number> = {
  equals: (x, y) => x === y,
};

export const eqPoint: eq.Eq<Point2D> = array.getEq(eqNumber);
```

and this complains that a `number[]` can't be assigned to Vec<2, number>. Is there a particular reason why this library doesn't seem to implement equality for the vectors?

**Comment by m-bock** [comment]
Hi!

The `Vec` type is implemented as readonly array. Just tried your code snippet and it works if you `import * as array from "fp-ts/ReadonlyArray"`

No, there was no specific reason to not to include a `getEq` into the library. I guess some others are missing, too.

I think it would be something like this:

```ts
const getEq: <A>(E: Eq<A>) => <N extends number>() => Eq<Vec<N, A>> = (
  E
) => () => array.getEq(E)
```

Thanks for posting this. PR's are always welcome :)

**Comment by m-bock** [comment]
Also created this one:
https://github.com/no-day/fp-ts-number-instances/issues/1

**Comment by justin-hackin** [comment]
Thanks for letting me know about read-only array feature. I decided that I only needed one size and thus I opted out of this library for now. Glad it was worth marking this in the number library.

**Comment by m-bock** [comment]
Yes, this makes sense. I actually had in mind to create a library that provides the fixed sized vectors. (Vec2<A>, Vec3<A>).
What exists though is one that has the non-generic `Vec2n`, `Vec3n` types:
https://github.com/no-day/fp-ts-numeric-vectors (unfortunately `Eq` and other instances missing, too)

In future there should be easy conversion between those types by adding `fromTuple`/`toTuple` everywere.
